miniconf/src/key.rs

```diff
@@ -1,53 +1,81 @@
-use crate::{Traversal, TreeKey};
+use crate::Traversal;
+
+/// The capability to look up top level field names and convert to indices
+///
+/// This trait is derived together with [`crate::TreeKey`].
+pub trait TreeLookup {
```

Comment:
I was expecting to find this in `tree.rs` given the name of the trait

Reply:
Interestingly this turned out to not be much like the other traits in `tree`. `TreeLookup` isn't depth-parametric and it's strictly only single (top) level len and name_to_index.

Reply:
Yeah I noticed that as well. Just mainly threw me off because it has the `Tree` prefix on the trait name.

Reply:
Ack. Let's call it `KeyLookup`.